src/helper_functions.rs

```diff
@@ -0,0 +1,200 @@
+            }
+        } else {
+            panic!("It's not reveal stage")
+        }
+    }
+
+    pub(crate) fn end_round(&mut self) -> RevealResult {
+        let set_of_moves = BTreeSet::from_iter(self.player_moves.values().cloned());
+        let next_round_players: BTreeSet<ActorId> = match set_of_moves.len() {
+            1 | 4 | 5 => self.player_moves.keys().cloned().collect(),
+            2 | 3 => {
+                let winners = self.next_round_moves_set(set_of_moves);
+                self.player_moves
+                    .iter()
+                    .filter(|(_, users_move)| winners.contains(users_move))
+                    .map(|(player, _)| player)
+                    .copied()
+                    .collect()
+            }
+            _ => panic!("Unknown result"),
+        };
+
+        if next_round_players.len() > 1 {
+            self.stage = GameStage::InProgress(StageDescription {
+                anticipated_players: next_round_players.clone(),
+                finished_players: BTreeSet::new(),
+            });
+            self.update_timestamp();
+            self.clear_moves();
+
+            RevealResult::NextRoundStarted {
+                players: next_round_players,
+            }
+        } else {
+            let winner = next_round_players
+                .into_iter()
+                .last()
+                .expect("Unknown winner");
+            msg::send(winner, "WIN", exec::value_available()).expect("Can't send reward");
+            self.start_new_game();
+
+            RevealResult::GameOver { winner }
+        }
+    }
+
+    pub(crate) fn next_round_moves_set(&self, set_of_moves: BTreeSet<Move>) -> BTreeSet<Move> {
+        'outer: for a_move in set_of_moves.iter() {
+            for b_move in set_of_moves.iter() {
+                if a_move != b_move && !a_move.wins(b_move) {
+                    continue 'outer;
+                }
+            }
+
+            return BTreeSet::from([a_move.clone()]);
+        }
+
+        set_of_moves
+    }
+
+    pub(crate) fn save_move(&mut self, player: &ActorId, move_hash: String) {
+        if let GameStage::InProgress(progress_description) = &mut self.stage {
+            self.encrypted_moves.insert(*player, move_hash);
+
+            progress_description.anticipated_players.remove(player);
+            progress_description.finished_players.insert(*player);
+        }
+    }
+
+    pub(crate) fn save_real_move(&mut self, player: &ActorId, real_move: &str) {
+        let users_move = real_move.chars().next().expect("Move is empty");
+        let users_move = Move::new(users_move);
+
+        self.player_moves.insert(*player, users_move);
+
+        match &mut self.stage {
+            GameStage::Preparation | GameStage::InProgress(_) => {}
+            GameStage::Reveal(description) => {
+                description.anticipated_players.remove(player);
+                description.finished_players.insert(*player);
+            }
+        }
```

Comment:
Better to underline the entity is needed to be mutable.
```suggestion
        match self.stage {
            GameStage::Preparation | GameStage::InProgress(_) => {}
            GameStage::Reveal(ref mut description) => {
                description.anticipated_players.remove(player);
                description.finished_players.insert(*player);
            }
        }
```